fastestimator/backend/get_gradient.py

```diff
@@ -77,8 +77,23 @@ def get_gradient(target: Tensor,
                                         retain_graph=retain_graph,
                                         create_graph=higher_order,
                                         only_inputs=True)
-        if isinstance(gradients, Tuple) and len(gradients) == 1:
-            # PyTorch seems to unnecessarily wrap results into a tuple when a non-scalar target is provided
+
+        if isinstance(gradients, Tuple) and len(gradients) == 1 and isinstance(sources, torch.Tensor):
```

Comment:
do we still need to check that len(gradients) == 1 if we're checking that sources is a torch.Tensor?

Reply:
I was considering you may see some cases that I didn't, so I just added an additional condition.
Since you also think that is unnecessary then I will simplify it.

Reply:
In that sense, I might also discard "isinstance(gradients, Tuple)" condition

Reply:
so long as your code comments about possible combinations is exhaustive. have you run pytorch saliency with these changes?

Reply:
No, but i have run nightly-test and it passed

Reply:
Just ran the xai pytorch version that was modified from t08_xai.ipynb and it works